sessionctx/variable/sysvar.go

```diff
@@ -1245,10 +1245,7 @@ var defaultSysVars = []*SysVar{
 		return nil
 	}},
 	/* The following variable is defined as session scope but is actually server scope. */
-	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableDynamicPrivileges, Value: Off, Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
-		s.EnableDynamicPrivileges = TiDBOptOn(val)
-		return nil
-	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableDynamicPrivileges, Value: On, Type: TypeBool, Hidden: true},
```

Comment:
This is just changed to hidden for a small amount of time so that the tidb-tests and tipocket tests that try to set this will still work.

Reply:
Do we want to remove it in the future?

Reply:
I plan to remove it. If I remove it right now though, the tidb-test and tipocket tests will fail (since they attempt to set it).

I will instead merge this, fix the tests, and then create a small followup PR.